pkg/vpcmodel/explainabilityConnectivity.go

```diff
@@ -135,79 +133,77 @@ func (c *VPCConfig) computeExplainRules(srcNodes, dstNodes []Node,
 	return rulesAndConn, nil
 }
 
-// computeAdditionalDetails computes, after potentialRules were computed, for each  <src, dst> :
+// computeFilters computes for each  <src, dst> :
 // 1. The routingResource
-// 2. The actual filters relevant to the src, dst given the routingResource
-// 3. from the potentialRules the actualRules (per ingress, egress) that actually enable traffic,
-// considering filtersExternal (which was computed based on the RoutingResource) and removing filters
-// not relevant for the Router e.g. nacl not relevant fip
-// 4. ingressEnabled and egressEnabled: whether traffic is enabled via ingress, egress
-
-func (details *rulesAndConnDetails) computeAdditionalDetails(c *VPCConfig) error {
+// 2. The external filters relevant to the <src, dst> given the routingResource
+// 3. The internal filters relevant to the <src, dst>
+// 4. The actual relevant filter, depending on whether src xor dst is external
+func (details *rulesAndConnDetails) computeFilters(c *VPCConfig) error {
 	for _, singleSrcDstDetails := range *details {
-		src := singleSrcDstDetails.src
-		dst := singleSrcDstDetails.dst
 		// RoutingResources are computed by the parser for []Nodes of the VPC,
 		// finds the relevant nodes for the query's src and dst;
 		// if for src or dst no containing node was found, there is no router
-		var routingResource RoutingResource
-		var filtersForExternal map[string]bool
-		var err error
-
-		routingResource, _, err = c.getRoutingResource(src, dst)
-		if err != nil {
-			return err
-		}
-		if routingResource != nil {
-			filtersForExternal = routingResource.AppliedFiltersKinds() // relevant filtersExternal
+		src := singleSrcDstDetails.src
+		dst := singleSrcDstDetails.dst
+		if src.IsInternal() && dst.IsInternal() { // internal
+			// security group applied always between two vsis
+			singleSrcDstDetails.filtersRelevant = map[string]bool{SecurityGroupLayer: true}
+			// nacl applied between two vsis if not same subnet
+			singleSrcDstDetails.filtersRelevant[NaclLayer] =
+				src.(InternalNodeIntf).Subnet().UID() != dst.(InternalNodeIntf).Subnet().UID()
```

Comment:
This logic is part of the model.. maybe should have a function `AppliedFiltersKinds()` for a pair of internal nodes? 
It can be added as another method to `InternalNode` interface , such as `AppliedFiltersKinds(otherNode InternalNode)`.
I think it is better have this as part of the model and not hard coded in this specific place. Might be relevant for other modules as well.

Reply:
right. done.